tests/common/dualtor/dual_tor_io.py

```diff
@@ -356,32 +359,68 @@ def traffic_sender_thread(self):
         # Signal data_plane_utils that sender and sniffer threads have begun
         self.io_ready_event.set()
 
+        sent_packets_count = 0
         for entry in self.packets_list:
+            _, packet = entry
+            server_addr = self.get_server_address(scapyall.Ether(str(packet)))
             time.sleep(self.send_interval)
+            # the stop_early flag can be set to True by data_plane_utils to stop prematurely
+            if self.stop_early:
+                self.stop_sniffer_early()
+                logger.info("Stop the sender thread gracefully after sending {} packets"\
+                    .format(sent_packets_count))
+                break
             testutils.send_packet(self.ptfadapter, *entry)
+            self.packets_sent_per_server[server_addr] =\
+                self.packets_sent_per_server.get(server_addr, 0) + 1
+            sent_packets_count = sent_packets_count + 1
 
         logger.info("Sender finished running after {}".format(
             str(datetime.datetime.now() - sender_start)))
 
 
+    def stop_sniffer_early(self):
+        # Try to stop sniffer earlier by sending SIGINT signal to the sniffer process
+        # Python installs a small number of signal handlers by default.
+        # SIGINT is translated into a KeyboardInterrupt exception.
+        logger.info("Stop the sniffer thread gracefully: sending SIGINT to ptf process")
+        sniffer_pid = self.ptfhost.command("pgrep -f {}".format(self.ptf_sniffer),\
```

Comment:
we can use `pkill` directly to avoid separate calls to `pgrep` and `kill`

Reply:
Done